app/services/degrees/dfe_reference.rb

```diff
@@ -2,35 +2,73 @@
 
 module Degrees
   class DfeReference
-    class << self
-      include MappingsHelper
+    OTHER = "Other"
+
+    COMMON_TYPES = ["Bachelor of Arts", "Bachelor of Science", "Master of Arts", "PhD"].freeze
+
+    SUPPORTED_GRADES_BY_HESA_CODES = %w[1 2 3 5 14].freeze
+
+    SUPPORTED_GRADES = DfE::ReferenceData::Degrees::GRADES.all_as_hash.select { |_, item|
+      SUPPORTED_GRADES_BY_HESA_CODES.include?(item[:hesa_code])
+    }
 
+    GRADES = DfE::ReferenceData::HardcodedReferenceList.new(
+      SUPPORTED_GRADES.merge(
+        {
+          "4182ef37-df1c-4f1e-9be6-feb66037f775" => {
+            name: OTHER,
+            match_synonyms: [],
+            suggestion_synonyms: [],
+          },
+        },
+      ),
+    )
+
+    INSTITUTIONS = DfE::ReferenceData::TweakedReferenceList.new(
+      DfE::ReferenceData::Degrees::INSTITUTIONS,
+      DfE::ReferenceData::Record.new(
+        {
+          "96e9359f-dbad-4486-8de9-f05f3c7104c2" => {
+            name: OTHER,
+            match_synonyms: [],
+            suggestion_synonyms: [],
+            abbreviation: nil,
+          },
+        },
+      ),
+    )
+
+    SUBJECTS = DfE::ReferenceData::Degrees::SINGLE_SUBJECTS
+    TYPES = DfE::ReferenceData::Degrees::TYPES_INCLUDING_GENERICS
+
+    class << self
       def find_subject(uuid: nil, name: nil, hecos_code: nil)
-        find_item(:single_subjects, id: uuid, hecos_code: sanitised_hesa(hecos_code), name: name)
+        find_item(:single_subjects, id: uuid, hecos_code: hecos_code, name: name)
       end
 
-      def find_type(uuid: nil, abbreviation: nil, hesa_code: nil)
+      def find_type(uuid: nil, name: nil, abbreviation: nil, hesa_code: nil)
         find_item(:types_including_generics,
                   id: uuid,
-                  hesa_itt_code: sanitised_hesa(hesa_code),
+                  name: name,
+                  hesa_itt_code: hesa_code,
                   abbreviation: abbreviation)
       end
 
       def find_institution(uuid: nil, name: nil, hesa_code: nil)
-        find_item(:institutions, id: uuid, hesa_itt_code: sanitised_hesa(hesa_code), name: name)
+        find_item(:institutions, id: uuid, hesa_itt_code: hesa_code, name: name)
       end
 
       def find_grade(uuid: nil, name: nil, hesa_code: nil)
-        find_item(:grades, id: uuid, hesa_code: sanitised_hesa(hesa_code), name: name)
+        find_item(:grades, id: uuid, hesa_code: hesa_code, name: name)
       end
 
       def find_item(list_type, filters)
         ref_dataset = DfE::ReferenceData::Degrees.const_get(list_type.to_s.upcase)
 
         hesa_filter = {
-          hesa_itt_code: filters[:hesa_itt_code],
-          hesa_code: filters[:hesa_code],
-          hecos_code: filters[:hecos_code],
+          hesa_itt_code: sanitised_hesa_code(filters[:hesa_itt_code]),
+          hesa_code: sanitised_hesa_code(filters[:hesa_code]),
+          hecos_code: sanitised_hesa_code(filters[:hecos_code]),
         }.compact
```

Comment:
nit: and optional, this could be extracted out into a private method to reduce the complexity of the `find_item` method.